angr/analyses/decompiler/optimization_passes/flip_boolean_cmp.py

```diff
@@ -0,0 +1,43 @@
+from typing import List
+
+from ailment.expression import BinaryOp, UnaryOp, Op
+
+from ..structuring.structurer_nodes import ConditionNode
+from ..structured_codegen.c import is_simple_return_node
+from .optimization_pass import SequenceOptimizationPass, OptimizationPassStage
+
+
+class FlipBooleanCmp(SequenceOptimizationPass):
+    """
+    In the scenario in which a false node has no apparent successors, flip the condition on that if-stmt.
+    This is only useful when StructuredCodeGenerator has simplify_else_scopes enabled, as this will allow the
+    flipped if-stmt to remove the redundant else.
+    """
+
+    ARCHES = ["X86", "AMD64"]
```

Comment:
Why are we limiting architectures to x86 and amd64? Also why are we limiting platforms? This seems to be a generic optimization for all architectures and platforms.